homeassistant/components/subaru/sensor.py

```diff
@@ -270,16 +269,22 @@ def available(self):
         last_update_success = super().available
         if last_update_success and self.vin not in self.coordinator.data:
             return False
+        if self.state is None:
+            return False
         return last_update_success
 
     def get_current_value(self):
         """Get raw value from the coordinator."""
-        value = self.coordinator.data[self.vin][VEHICLE_STATUS].get(self.data_field)
-        if value in sc.BAD_SENSOR_VALUES:
-            value = None
-        if isinstance(value, str):
-            if "." in value:
-                value = float(value)
-            else:
-                value = int(value)
-        return value
+        if isinstance(self.coordinator.data, dict):
+            value = self.coordinator.data.get(self.vin)[VEHICLE_STATUS].get(
+                self.data_field
+            )
```

Comment:
Would this style work?
```suggestion
        if isinstance(data := self.coordinator.data, dict):
            value = data.get(self.vin)[VEHICLE_STATUS].get(self.data_field)
```

Reply:
yes, thanks!